### DISCUSSION TITLE
Need help with autograding HTML files.

### DISCUSSION THREAD

**Discussion by marionlester** [community/community#49596] [GitHub Education] [answered]
Hello. I'm a teaching html to my class and need some help setting up autograding my assignments. I'm (very) new to github and github classroom so please be patient with me. 

What I need is a simple check for erroneous tags and I experimented with the Input/Output test as shown in the official Github Classroom youtube video. I just followed it and hoped for the best (I used npm install and npm test).  I have no idea about what to type on the setup/run command field, and I guess that's the reason why I'm stuck on my issue. There are also some errors as indicated on the picture. 

[https://drive.google.com/file/d/122ADDeQaEyHt5zTfKqaBhpR1B6mMzL0D/view?usp=share_link](url)
[https://drive.google.com/file/d/1YAUuEltic0tpC65AMnwrciDeDbm9H59D/view?usp=share_link](url)

I have around 80 students in my laboratory, so autograding will definitely help. Thanks in advance.

**Comment by jge162** [comment, marked as answer]
A basic JavaScript you can use is this.

```javascript

const fs = require('fs');
const path = require('path');
const { JSDOM } = require('jsdom');

test('no erroneous tags', () => {
  const filename = 'index.html'; // make sure the HTML file is named index.html
  const fileContent = fs.readFileSync(filename, 'utf8');
  const dom = new JSDOM(fileContent);
  const erroneousTags = dom.window.document.querySelectorAll('marquee, blink'); // test cases here. 
  expect(erroneousTags.length).toBe(0);
});
```
this example, the test checks that there are no  `<marquee>` or `<blink>` tags in the HTML file. You can modify the test as needed to check for other errors that suit your use case. 

Bests.

**Reply by marionlester** [reply]
wow thanks.. i'll check this out.

**Reply by marionlester** [reply]
Thanksssss... It's working now!!!

**Comment by marionlester** [comment]
Thanks for the additional info. I was currently giving manual feedback through the feedback pull feature. It takes too much time tho. Now I'm able to proceed on focusing on teaching.

**Reply by jge162** [reply]
Glad to hear. Bests

**Reply by marionlester** [reply]
I got a question, can we hide the package.json and test.js files? this is kinda bit a security issue since my students can be very curious on the files on their own cloned repos. As I've read this is still a feature request right?

**Reply by jge162** [reply]
I believe it is still a feature request at this time. Also you can use a private repo but not sure your specific use case and if a private repo is applicable.
